src/Model/Item.php

```diff
@@ -1445,7 +1445,24 @@ public static function isValid(array $item)
 			Logger::notice('Causer is ignored by user', ['causer-link' => $item['causer-link'], 'uid' => $item['uid'], 'item-uri' => $item['uri']]);
 			return false;
 		}
-		
+
+		$toplevelOwnerId = null;
+		if ($item['parent-uri'] != $item['uri']) {
+			$topLevelUriId   = ItemURI::getIdByURI($item['parent-uri']);
+			$toplevelItem    = self::selectFirst(['owner-link'], ['uri-id' => $topLevelUriId]);
+			$toplevelOwnerId = User::getIdForURL($toplevelItem['owner-link'] ?? '');
+		}
+
+		if (!empty($toplevelOwnerId) && !empty($item['author-id']) && Contact\User::isBlocked($item['author-id'], $toplevelOwnerId)) {
+			Logger::notice('Author is blocked by parent user', ['author-link' => $item['author-link'], 'uid' => $toplevelOwnerId, 'item-uri' => $item['uri']]);
+			return false;
+		}
+
+		if (!empty($toplevelOwnerId) && !empty($item['owner-id']) && Contact\User::isBlocked($item['owner-id'], $toplevelOwnerId)) {
+			Logger::notice('Owner is blocked by parent user', ['owner-link' => $item['owner-link'], 'uid' => $toplevelOwnerId, 'item-uri' => $item['uri']]);
+			return false;
+		}
+
```

Comment:
Please also check for the `causer-id`.

Reply:
The causer is only blocked by the user if the item is top-level, but the new check is specifically about replies, so I didn't think it was relevant.

Reply:
A check for the causer wouldn't hurt. And I would not exclude some situation where the causer could contain a blocked person as well.